src/borg/testsuite/compress.py

```diff
@@ -11,36 +11,39 @@
 
 from ..compress import get_compressor, Compressor, CompressionSpec, CNONE, ZLIB, LZ4, LZMA, ZSTD, Auto
 
-
-buffer = bytes(2**16)
 data = b"fooooooooobaaaaaaaar" * 10
 params = dict(name="zlib", level=6)
 
 
-def test_get_compressor():
-    c = get_compressor(name="none")
-    assert isinstance(c, CNONE)
-    c = get_compressor(name="lz4")
-    assert isinstance(c, LZ4)
-    c = get_compressor(name="zlib")
-    assert isinstance(c, ZLIB)
-    with pytest.raises(KeyError):
-        get_compressor(name="foobar")
-
-
-def test_cnull():
-    c = get_compressor(name="none")
+@pytest.mark.parametrize(
+    "c_type, expected_compressor",
+    [
+        ("none", CNONE),
+        ("lz4", LZ4),
+        ("zlib", ZLIB),
+        ("foobar", None),
+    ],
+)
+def test_get_compressor(c_type, expected_compressor):
+    if expected_compressor is not None:
+        compressor = get_compressor(name=c_type)
+        assert isinstance(compressor, expected_compressor)
+    else:
+        with pytest.raises(KeyError):
+            get_compressor(name=c_type)
+
+
+@pytest.mark.parametrize(
+    "c_type",
+    ['none', 'lz4', 'zlib', 'zstd'] + (['lzma'] if lzma else [])
```

Comment:
just assume lzma is always there (and also simplify the import).

guess many years ago i wanted the tests to be able to work even with half-broken self-compiled pythons that miss lzma support, but considering lzma is a documented part of the python standard library, i guess we can get rid of that.